innovation.game.php

```diff
@@ -371,6 +371,32 @@ function debug_dig($card_id) {
             throw new BgaUserException(self::format("This card is in {player_name}'s {location}", array('player_name' => self::getPlayerNameFromId($card['owner']), 'location' => $card['location'])));
         }
     }
+    function debug_foreshadow($card_id) {
+        if (self::getGameStateValue('debug_mode') == 0) {
```

Comment:
This is already the case today:
```
if (gamedatas.echoes_expansion_enabled) {
                    main_area.innerHTML = "<button id='debug_foreshadow' class='action-button debug_button bgabutton bgabutton_red'>FORESHADOW</button>" + main_area.innerHTML;
                }
                if (gamedatas.artifacts_expansion_enabled) {
                    main_area.innerHTML = "<button id='debug_dig' class='action-button debug_button bgabutton bgabutton_red'>DIG</button>" + main_area.innerHTML;
                }
```

I'm not concerned about enforcing it on the backend since there's no security risk in the studio, and these actions can't be called in production.